source/opt/instruction.h

```diff
@@ -162,7 +162,8 @@ class Instruction : public utils::IntrusiveNodeBase<Instruction> {
         has_type_id_(false),
         has_result_id_(false),
         unique_id_(0),
-        dbg_scope_(kNoDebugScope, kNoInlinedAt) {}
+        dbg_scope_(kNoDebugScope, kNoInlinedAt),
+        is_cl100_dbg_inst_(false) {}
```

Comment:
I don't like having this.  We are burning 32-bit for every instruction, and all that we gain is to avoid checking that this is an instruction in the appropriate extended instruction set.  We can cache the id of the cl100_dbg extended instruction set like we do for the glslstd450 (https://github.com/KhronosGroup/SPIRV-Tools/blob/master/source/opt/feature_manager.h#L86).